toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -34,20 +52,44 @@ func extractPartitions(imageLoopDevice string, outDir string, basename string, p
 			case "raw":
 				// Do nothing for "raw" case.
 			case "raw-zst":
-				partitionFilepath, err = compressWithZstd(partitionFilepath)
+				partitionFilepath, err = extractRawZstPartition(partitionFilepath, skippableFrameMetadata, partitionfileName, outDir)
 				if err != nil {
 					return err
 				}
 			default:
 				return fmt.Errorf("unsupported partition format (supported: raw, raw-zst): %s", partitionFormat)
 			}
-
 			logger.Log.Infof("Partition file created: %s", partitionFilepath)
 		}
 	}
 	return nil
 }
 
+// Extract raw-zst partition
+func extractRawZstPartition(partitionRawFilepath string, skippableFrameMetadata [SkippableFramePayloadSize]byte, partitionfileName string, outDir string) (partitionFilepath string, err error) {
+	// Compress raw partition with zstd
+	tempPartitionFilepath, err := compressWithZstd(partitionRawFilepath)
```

Comment:
where is partitionRawFilepath coming from?

Reply:
It is defined in line 46 - it is the filepath of the raw partition file